core/src/main/java/jenkins/install/SetupWizard.java

```diff
@@ -0,0 +1,183 @@
+package jenkins.install;
+
+import java.io.IOException;
+import java.util.Locale;
+import java.util.UUID;
+import java.util.logging.Logger;
+
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletRequestWrapper;
+
+import org.kohsuke.stapler.HttpResponse;
+import org.kohsuke.stapler.StaplerRequest;
+import org.kohsuke.stapler.StaplerResponse;
+
+import hudson.BulkChange;
+import hudson.ExtensionList;
+import hudson.model.User;
+import hudson.model.UserProperty;
+import hudson.security.FullControlOnceLoggedInAuthorizationStrategy;
+import hudson.security.HudsonPrivateSecurityRealm;
+import hudson.security.PermissionAdder;
+import hudson.security.SecurityRealm;
+import hudson.security.csrf.DefaultCrumbIssuer;
+import hudson.util.HttpResponses;
+import hudson.util.PluginServletFilter;
+import jenkins.model.Jenkins;
+import jenkins.security.s2m.AdminWhitelistRule;
+
+/**
+ * A Jenkins instance used during first-run to provide a limited set of services while
+ * initial installation is in progress
+ */
+public class SetupWizard {
+    /**
+     * The security token parameter name
+     */
+    public static String initialSetupAdminUserName = "admin";
+
+    private final Logger LOGGER = Logger.getLogger(SetupWizard.class.getName());
+
+    public SetupWizard(Jenkins j) throws IOException {
+        User admin;
+        // Create an admin user by default with a 
+        // difficult password
+        if(j.getSecurityRealm() == null || j.getSecurityRealm() == SecurityRealm.NO_AUTHENTICATION) { // this seems very fragile
+            BulkChange bc = new BulkChange(j);
+            
+            HudsonPrivateSecurityRealm securityRealm = new HudsonPrivateSecurityRealm(false, false, null);
+            j.setSecurityRealm(securityRealm);
+            String randomUUID = UUID.randomUUID().toString().replace("-", "").toLowerCase(Locale.ENGLISH);
+            admin = securityRealm.createAccount(SetupWizard.initialSetupAdminUserName, randomUUID);
+            admin.addProperty(new SetupWizard.AuthenticationKey(randomUUID));
+            
+            // Lock Jenkins down:
+            FullControlOnceLoggedInAuthorizationStrategy authStrategy = new FullControlOnceLoggedInAuthorizationStrategy();
+            authStrategy.setAllowAnonymousRead(false);
+            j.setAuthorizationStrategy(authStrategy);
+
+            // there probably aren't any implementors yet, but just in case
+            for (PermissionAdder adder : ExtensionList.lookup(PermissionAdder.class)) {
+                adder.add(authStrategy, admin, Jenkins.ADMINISTER);
```

Comment:
Given that you are ignoring the return value, this extension point is not being used as intended. Unless you have a reason to call it, better IMO just to delete this block.